### ISSUE TITLE
tipsy front end not converted to yt4.x

### ISSUE THREAD

**Issue by dnarayanan** [dnarayanan/powderday#49]
the tipsy front end does not yet have the machinery in it to work for yt4.x.  look to gadget2pd for examples of where to add the relevant if statements.

**Comment by dnarayanan** [comment]
this might not be possible yet. it looks like accessing octree information in yt4.x via (e.g.) `octree[('gas','density')]` isn't working.  but maybe i'm misunderstanding something about how to access these fields for tipsy setups.

**Comment by anchwr** [comment]
Any updates? I've downloaded Ash Kelly's octree branch of yt and reverted a couple of powderday files back to old versions to get powderday running with tipsy outputs from Gasoline. It would be nice to not have to maintain multiple installations of yt and powderday, but I also have no idea how to access the octree information in yt4.x.

**Comment by anchwr** [comment]
Looks like [this feature](https://yt-project.org/doc/yt4differences.html#smoothing-data-onto-an-octree) might allow the tipsy frontend to be updated.

**Comment by dnarayanan** [comment]
hi sorry for the delay - I've been on travel all week!  but yeah - this is the nomenclature for accessing the octree information.

I actually think this would be pretty easy -- I think its going to be along the lines of replacing tings like:

```    
def _gassmootheddensity(field,data):
        return data[('deposit', 'Gas_density')]
 ```
 
 with
 
 ```
     def _gassmootheddensity(field,data):
        return octree['PartType0','density']
        ```
        
 I should hopefully be able to play with it next week when I'm back to try to fix it up but if you want to give it a try before then, would very much welcome a PR!

**Comment by dnarayanan** [comment]
(if you look in the gadget front end and search for `octree` you'll see some examples of how to do this.  I _think_ this is the only real update needed for yt4.x

also you don't need to be on ash Kelly's branch of `yt` anymore !  the tip of main should be good now.   (I just updated the docs for this)

**Comment by anchwr** [comment]
I might be able to give it a shot this weekend! And yeah - I have a typical yt installation for Enzo stuff, but, since the powderday tipsy frontend still depends on the old octree implementation and I knew Ash Kelly's branch still had that, I downloaded it to do some Gasoline test runs.

**Comment by dnarayanan** [comment]
okay awesome - if you have troubles feel free to post here!

**Comment by anchwr** [comment]
Thanks for the suggestion - the gadget2py functions were very useful! I just created the pull request. I also (I hope!) updated tipsy2pd to allow the li_ml dust grid type to be used. The images and SEDs that I've run with this+tipsy outputs look reasonable, but let me know if the implementation looks wrong! I based most of the changes on the gadget2py functions, but also added some calculations to _gassfr to estimate the instantaneous gas SFR. It largely grabs the star formation criteria from the simulation's parameters, but I couldn't figure out how to grab the density cut that most Gasoline sims use - I've put in the value that's used for a lot of them, but this is something users might have to change if they're using a simulation with a different setting and want to use the li_ml dust grid.

**Comment by dnarayanan** [comment]
this was fixed in commit e94e86b7bcc7de6d0284cb4d7798a0bcead4f443 by @anchwr -- thanks for the nice PR!
